daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -357,7 +349,7 @@ private[speedy] case class PartialTransaction(
       stakeholders: Set[Party],
       key: Option[Node.KeyWithMaintainers],
       version: TxVersion,
-  ): (Value.ContractId, PartialTransaction) = {
+  ): Either[(PartialTransaction, Tx.TransactionError), (Value.ContractId, PartialTransaction)] = {
```

Comment:
When `insertCreate`, returns the `Left` case to signify failure, it also returns an updated ptx.
This seems very odd.
But one testcase in `ScriptService` will fail otherwise.